src/components/Filtering/FilteringService.ts

```diff
@@ -0,0 +1,159 @@
+    priorityTarget: '',
+  });
+
+  useEffect(() => {
+    // setStorage('todos', TODOS);
+    const getData = getStorage('todos') || [];
+
+    setOriginalData(getData);
+    setTodoState(getData);
+  }, []);
+
+  useEffect(() => {
+    setTodoState(todoState);
+  }, [todoState]);
+
+  // const handlerFiltering = (e: React.MouseEvent<HTMLButtonElement>): void => {
+  const handlerFiltering = (target: string): void => {
+    if (target === MENU.FILTER[0]) handlerOriginState();
+    if (target === MENU.FILTER[1]) handlerDeadlineState();
+    if (target === MENU.PRIORITY[0]) {
+      handlerPriorityState('low');
+      setOpen(!open);
+    }
+    if (target === MENU.PRIORITY[1]) {
+      handlerPriorityState('medium');
+      setOpen(!open);
+    }
+    if (target === MENU.PRIORITY[2]) {
+      handlerPriorityState('high');
+      setOpen(!open);
+    }
+  };
+
+  const handlerPriorityState = (target: string) => {
+    if (click.deadline) {
+      const priorityArr = filteringPriority(target);
+      const deadlineArr = filteringDeadline();
+      const intersectionArr = priorityArr.filter(arr =>
+        deadlineArr.includes(arr)
+      );
+
+      setClick({
+        ...click,
+        origin: false,
+        priority: true,
+        priorityTarget: target,
+      });
+      setTodoState(intersectionArr);
+    } else {
+      setClick({
+        ...click,
+        origin: false,
+        priority: true,
+        priorityTarget: target,
+      });
+      setTodoState(filteringPriority(target));
+    }
+  };
+
+  const handlerDeadlineState = () => {
+    if (open) setOpen(!open);
+
+    if (click.deadline) {
+      setClick({ ...click, origin: true, deadline: false });
+      setTodoState(originalData);
+    }
+
+    if (click.priority) {
+      if (click.deadline) {
+        setClick({ ...click, origin: false, deadline: false });
+        setTodoState(filteringPriority(click.priorityTarget));
+      } else {
+        const priorityArr = filteringPriority(click.priorityTarget);
+        const deadlineArr = filteringDeadline();
+        const intersectionArr = priorityArr.filter(arr =>
+          deadlineArr.includes(arr)
+        );
+
+        setClick({ ...click, origin: false, deadline: true });
+        setTodoState(intersectionArr);
+      }
+    }
+    if (!click.deadline && !click.priority) {
+      setClick({ ...click, origin: false, deadline: true });
+      setTodoState(filteringDeadline());
+    }
+  };
+
+  const handlerOriginState = () => {
+    if (open) setOpen(!open);
+    setClick({
+      ...click,
+      origin: true,
+      priority: false,
+      deadline: false,
+      priorityTarget: 'Priority',
+    });
+    setTodoState(originalData);
+  };
+
+  const handlerDrop = (): void => {
+    setOpen(!open);
+  };
+
+  const filteringPriority = (target: string): Itodo[] => {
+    const data = originalData;
+    const filtering = data.filter((todo: Itodo) => todo.priority === target);
+
+    return filtering;
+  };
```

Comment:
[질문] 이 친구가! 우선순위 (로우 미디엄 하이) 중에 골랐을때, 그 고른 우선순위에 해당하는 todo를 가지게 하는? 친구인건가요!!

만약 맞다면) 132번째 줄에서 `filteredData` 나, `filteredTodos` , 가 더 좋지 않을까 생각이 들어요!

아 함수명도! 다음처럼은 어떨까요?

```suggestion
  const filterTodoByPriority= (priorityStd: string): Itodo[] => {
    const data = originalData;
    const filteredTodos= data.filter((todo: Itodo) => todo.priority === priorityStd);

    return filteredTodos;
  };
```

Reply:
맞습니다. 의견 주신대로 함수명을 명시적으로 바꾸겠습니다.